lib/trip_updates/trip_updates_pub_sub.ex

```diff
@@ -108,10 +108,18 @@ defmodule TripUpdates.TripUpdatesPubSub do
 
   def is_it_fresh?(trip_update, current_time \\ DateTime.now!("America/New_York")) do
     # returns true if most recent arrival time in a stop update is in the future
-    last_stop_arrival(trip_update) >= current_time
+    with %DateTime{} = last_stop <- last_stop_arrival(trip_update),
+         :gt <- DateTime.compare(last_stop, current_time) do
+      true
+    else
+      _ -> false
+    end
   end
 
   defp last_stop_arrival(trip_update) do
-    trip_update.stop_time_update |> Enum.map(& &1.arrival_time) |> Enum.max(DateTime)
+    trip_update.stop_time_update
+    |> Enum.map(& &1.arrival_time)
+    |> Enum.filter(&(!is_nil(&1)))
```

Comment:
suggestion (non-blocking): There's also `Enum.reject` if you wanted to avoid the `!`.